lesson/outline/rtl-outline-template.md

```diff
@@ -0,0 +1,330 @@
+2. Learners will be able to make a component available in a test using RTL's `render` function and check for its availability using `screen.debug()`.
+
+### Narrative Summary
+
+_Check out the [content standards](http://curriculum-documentation.codecademy.com/Content-Standards/narrative/) for guidance on writing narratives for exercises._
+
+1. Introduce Jest as the testing framework that will be used to conduct our unit tests. It can be installed using the command `npm install --save-dev jest`.
+1. In order to use React Testing Library learners need to install it using the command `npm install --save-dev @testing-library/react`.
+3. React Testing Library has a `render` function that makes the component available in the unit test. `render` can be imported like so: `import { render } from '@testing-library/react'`. Emphasize that `render` takes in JSX as the argument.
+
+```javascript
+render(<h1>Hello</h1>)
+```
+4. The availability of this component can be verified using the method `screen.debug()` which prints out all the DOM contents.
+5. `screen` is a special object which can be thought of as a representation of the browser window. It is a part of the `DOM-testing-library` (RTL is an extension of this library). `screen` can be imported like so: `import { screen } from '@testing-library/react'`.
+
+
+### Checkpoints Summary
+
+_Check out the [content standards](http://curriculum-documentation.codecademy.com/Content-Standards/checkpoint/) for guidance on writing narratives for exercises._
+
+1. Install and import `React Testing Library`.
+2. Import and call the `render` function to make the react component available in a unit test.
+2. Check the availability of this component using `screen.debug()`
+
+#### What is the purpose of these checkpoints?
+
+1. So that learners know how to set up RTL.
+2. So that learners can implenment the `render` function and make the intended component available in the unit test.
+
+
+### What would you like to have in the workspace for this exercise? Share your plan below.
+
+- Perhaps a simple React component with paragraph text. This is just so that learners can see how the `render` function works.
+
+<hr>
+
+## Exercise # 3: Querying with RTL
+
+### Which course outcomes will be covered by this exercise?
+
+1. Learners will be able to perform queries with RTL helper methods.
+2. Learners will understand the difference between different types of query methods.
+
+### Narrative Summary
+
+_Check out the [content standards](http://curriculum-documentation.codecademy.com/Content-Standards/narrative/) for guidance on writing narratives for exercises._
+
+1. Explain to learners that RTL has a bunch of useful helper methods that can be used to query for DOM elements.
+2. There are two ways to perform these queries.
+- Either by the helper methods returned by the `render` function.
+```javascript
+const { getByText } = render(<div>Baz</div>)
+expect(getByText('Baz')).toBeInTheDocument()
+```
```

Comment:
The second way is the preferred way: https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/prefer-screen-queries.md

We should only teach that